pkg/proxy/utils.go

```diff
@@ -40,11 +85,40 @@ func GetSchema(ctx context.Context, c client.Client, namespace string, name stri
 	return schema, nil
 }
 
-// GetSampleDataset generates a sample dataset based on the provided dataset name.
-func GetSampleDataset(ctx context.Context, c client.Client, namespace string, datasetName string) (string, *bytes.Buffer, error) {
+// AddFileToTar adds a file with the given content to a tar.Writer.
+func AddFileToTar(tw *tar.Writer, name string, content *[]byte) error {
+	header := &tar.Header{
+		Name: name,
+		Mode: 0600,
+		Size: int64(len(*content)),
+	}
+	if err := tw.WriteHeader(header); err != nil {
+		return err
+	}
+	_, err := tw.Write(*content)
+	return err
+}
+
+// AddFileToZip adds a file with the given content to a zip.Writer.
+func AddFileToZip(zw *zip.Writer, name string, content *[]byte) error {
```

Comment:
Why use `*[]byte` for the function signature?

Reply:
Fixed. The initial purpose of passing a pointer was to avoid copying and improve performance. However, since the essence of a slice is a pointer, it is redundant to pass a pointer of a slice.